### PR TITLE
Fixes for kano code mini remix functionality

### PR DESCRIPTION
**Pull request by charlie-kano**
Minor fixes to aid remix functionality in kano code mini.

- Added accessor for content-widgets
- Added onDidRequestNextChallenge to ChallengeBase as used in Briefing, Challenge and Remix.

### CODE REVIEW

**Comment by TeamKano** [comment]
null

*From Jenkins: kano-code/PR-1839*

File: src/app/lib/challenge/base.ts [hunk 1]
```diff
@@ -16,6 +16,9 @@ export abstract class ChallengeBase {
     protected _onDidRequestShare = new EventEmitter();
     get onDidRequestShare() { return this._onDidRequestShare.event; }
 
+    protected _onDidRequestNextChallenge = new EventEmitter();
+    get onDidRequestNextChallenge() { return this._onDidRequestNextChallenge.event; }
+
```
Comment (AliceMarshall) [inline src/app/lib/challenge/base.ts:21]: so much better located here

File: src/app/lib/editor/widget/content-widgets.ts [hunk 2]
```diff
@@ -66,11 +66,16 @@ export class ContentWidgets {
...
         if (this.domNode.contains(widget.getDomNode())) {
             this.domNode.removeChild(widget.getDomNode());
         }
         const idx = this.widgets.indexOf(widget);
         this.widgets.splice(idx, 1);
     }
-}
\ No newline at end of file
+
+    getWidgets(): IEditorWidget[]{
```
Comment (AliceMarshall) [inline src/app/lib/editor/widget/content-widgets.ts:78]: nice!

File: src/app/lib/remix/remix.ts [hunk 3]
```diff
@@ -78,5 +77,19 @@ export class Remix extends ChallengeBase {
                 });
         }
     }
-    dispose() {}
+
+    stop() {}
+
+    dispose() {
+        // teardown banner 
+        if (this.editor.contentWidgets) {
```
Comment (AliceMarshall) [inline src/app/lib/remix/remix.ts:85]: 👍

**Pull request merged**